nca/CoreDS/ConnectionSet.py

```diff
@@ -550,3 +571,110 @@ def get_non_tcp_connections():
         res.add_all_connections([ProtocolNameResolver.get_protocol_number('TCP')])
         return res
         # return ConnectionSet(True) - ConnectionSet.get_all_TCP_connections()
+
+    # TODO - after moving to the optimized HC set implementation,
+    #  get rid of ConnectionSet and move the code below to ConnectivityProperties.py
+    @staticmethod
+    def conn_props_to_fw_rules(conn_props, cluster_info, peer_container, ip_blocks_mask,
+                               connectivity_restriction):
+        """
+        Build FWRules from the given ConnectivityProperties
+        :param ConnectivityProperties conn_props: properties describing allowed connections
+        :param ClusterInfo cluster_info: the cluster info
+        :param PeerContainer peer_container: the peer container
+        :param IpBlock ip_blocks_mask: IpBlock containing all allowed ip values,
+         whereas all other values should be filtered out in the output
+        :param Union[str,None] connectivity_restriction: specify if connectivity is restricted to
+               TCP / non-TCP , or not
+        :return: FWRules map
+        """
+        ignore_protocols = ProtocolSet()
+        if connectivity_restriction:
+            if connectivity_restriction == 'TCP':
+                ignore_protocols.add_protocol('TCP')
+            else:  # connectivity_restriction == 'non-TCP'
+                ignore_protocols = ProtocolSet.get_non_tcp_protocols()
+
+        fw_rules_map = defaultdict(list)
+        for cube in conn_props:
+            conn_cube = conn_props.get_connectivity_cube(cube)
+            src_peers = conn_cube.get_dim("src_peers")
+            if not src_peers:
+                src_peers = peer_container.get_all_peers_group(True)
+            conn_cube.unset_dim("src_peers")
+            dst_peers = conn_cube.get_dim("dst_peers")
+            if not dst_peers:
+                dst_peers = peer_container.get_all_peers_group(True)
+            conn_cube.unset_dim("dst_peers")
+            if IpBlock.get_all_ips_block() != ip_blocks_mask:
+                src_peers.filter_ipv6_blocks(ip_blocks_mask)
+                dst_peers.filter_ipv6_blocks(ip_blocks_mask)
+            protocols = conn_cube.get_dim("protocols")
+            conn_cube.unset_dim("protocols")
+            if not conn_cube.has_active_dim() and (not protocols or protocols == ignore_protocols):
+                conns = ConnectionSet(True)
+            else:
+                conns = ConnectionSet()
```

Comment:
I do not understand the condition `not protocols` in the `if` statement.
It should be of type `ProtocolSet`. A cube should not have any empty dimension value.
And why do you create `ConnectionSet(True)` if `not protocols` is true?

Reply:
Right, this was a leftover from some older version. Removed `not protocols` from `if`.